### ISSUE TITLE
Build process generate strange javascript 

### ISSUE THREAD

**Issue by cojack** [drudge/passport-facebook-token#88]
Hi,
First of all, thanks @drudge for your work, but there is a problem with this builder library, I was trying to use it with https://github.com/nestjs/passport but passed parameters up above to the `OAuth2Strategy` was only the values from super:

```typescript
export class FacebookTokenStrategy extends PassportStrategy(Strategy) {
	constructor(private readonly authService: AuthService) {
		super({
			clientID: config.facebook.app_id, // this
			clientSecret: config.facebook.app_secret // and that
		});
	}

	async validate(accessToken: string, refreshToken: string, profile) {
		return profile;
	}
}
```

Without options from your constructor, so basically what I have done is just copy your code to my project and allowJs set to true in tsconfig.json and it starts working.

So basically, if you would like to spend additional effort on this library to improve build process or even make it better and rewrite it to typescript I will be glad.

Once again, thanks a lot for sharing your code.

Regards.
